iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -312,23 +326,37 @@ TEST_F(PortManager_test, InterfacesOverflow)
             });
 
         auto interfacePointer = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::INTERNAL, "itfPenguin");
-        EXPECT_THAT(interfacePointer, Eq(nullptr));
+        EXPECT_EQ(interfacePointer, nullptr);
         EXPECT_TRUE(errorHandlerCalled);
     }
+}
+
+TEST_F(PortManager_test, deleteInterfacePortfromMaximumNumberAndAddOneIsSuccessful)
+{
+    std::string itf = "itf";
+
+    for (unsigned int i = 0; i < iox::MAX_INTERFACE_NUMBER; i++)
+    {
+        auto newItfName = itf + std::to_string(i);
+        auto interp = m_portManager->acquireInterfacePortData(
+            iox::capro::Interfaces::INTERNAL, iox::ProcessName_t(iox::cxx::TruncateToCapacity, newItfName));
+        EXPECT_NE(interp, nullptr);
+    }
```

Comment:
This is also repeated quite often. If you like, you could put this in separate method and just call that one. Additionally, if the ports are needed for later use, the method could have an optional lambda which takes the port and could be used like this
```c++
std::vector<iox::popo::InterfacePortData*> interfaceContainer;
exhaustInterfacePorts([&](auto port){
    interfaceContainer.push(port);
});
```

Reply:
Hi @elBoberido, Do you mean something like this:

template typename vector

iox::popo::InterfacePortData* acquireInterfacePortData(vector& Container, std::string itf)
{
    for (auto item : Container)
    {
        auto newItfName = itf + std::to_string(item);
        auto interp = m_portManager->acquireInterfacePortData(
            iox::capro::Interfaces::INTERNAL, iox::ProcessName_t(iox::cxx::TruncateToCapacity, newItfName));
        ASSERT_NE(interp, nullptr);
    }
    return interp;
}

and in the test:

{
std::vector<iox::popo::InterfacePortData*> interfaceContainer;
    std::string itf = "itf";

    // first aquire all possible condition variables
    auto interp = acquireInterfacePortData(interfaceContainer, itf);
    auto exhaustInterfacePorts([&](auto interp){
    interfaceContainer.push(interp);});
...
i didn´t know how the template could take the optional lambda in this case.

Reply:
This would be without templates, just for the specific port, e.g.
```
void exhaustInterfacePorts(std::function<void(iox::popo::InterfacePortData*)> f = std::function<void(iox::popo::InterfacePortData*)>())
{
    for(...)
    {
        auto port = ...
        ASSERT ...
        if (f)
        {
            f(port);
        }
    }
}
```
This could then be used like I mentioned in the previous comment

Reply:
If I understand this is the implementation needed:

           void exhaustInterfacePorts(std::string itf , PortManagerTester* m_portManager, 
           std::function<void(iox::popo::InterfacePortData*)> f = std::function<void(iox::popo::InterfacePortData*)>())
           {
                  for (unsigned int i = 0; i < iox::MAX_INTERFACE_NUMBER; i++)
                  {
                       auto newItfName = itf + std::to_string(i);
                       auto port = m_portManager->acquireInterfacePortData(
                        iox::capro::Interfaces::INTERNAL, iox::ProcessName_t(iox::cxx::TruncateToCapacity, newItfName));
                        ASSERT_NE(port, nullptr);
                            if (f)
                            {
                                f(port);
                             }
                       }

                }
and in the test it will be:
               {
                      std::vector<iox::popo::InterfacePortData*> interfaceContainer;
                      std::string itf = "itf";

                       // first aquire all possible condition variables
                      exhaustInterfacePorts(itf, m_portManager,[&](auto port){
                      interfaceContainer.push_back(port);});
...
If yes this two tests also need two similar methods no?
AcquireConditionVariablesDataAfterDestroyingPreviouslyAcquiredOnesIsSuccessful, AcquireNodeDataAfterDestroyingPreviouslyAcquiredOnesIsSuccessful,

Reply:
`m_portManager` is already part of the fixture and doesn't need to be passed to the method.
The interface name could be passed, yes.
Yes, the same could be done for the other tests were a resource is exhausted and where it is done multiple times

Reply:
elBoberido, i have another question, how to deal with this expects in test AcquiringMaximumNumberOfNodesWorks:     
        EXPECT_THAT(nodeData.value()->m_node, StrEq(newNodeName));
        EXPECT_THAT(nodeData.value()->m_process, StrEq(newProcessName));
should i add it in the method related to node , since AcquireNodeDataAfterDestroyingPreviouslyAcquiredOnesIsSuccessful didn´t use those expects?

Reply:
@manel94 I'm quite busy at the moment and come back later, but regarding the process name issue you can check if it is possible to always ave variable names and use it that way. Alternatively you can create an enum, e.g.
```
enum class NamePostfix
{
    OFF,
    COUNTER,
};
```
This can be passed as parameter to `exhaustInterfacePorts`.
One idea might also be to let `exhaustInterfacePorts` return the number of successfully acquired ports. The assert cound then be done on that number instead of in `exhaustInterfacePorts`.

Reply:
@elBoberido could you please explain more, i didn´t get you, i didn´t have any problem with exhaustInterfacePorts, my question is  how to deal with this test  AcquiringMaximumNumberOfNodesWorks to use exhaustNodePorts.

Reply:
@manel94 that was meant for the issue with the static and variable names. There is another PR which extends the tests for this file and there the method is called `acquireMaxNumberOfEventVariables`, so it might be a good idea to follow that naming.
So this would become `acquireMaxNumberOfNodePorts`. The `ASSERT_*` would only be necessary for `EXPECT_THAT(nodeData.has_error(), Eq(false));` the other ones can be changed but are not necessary since it doesn't have an potential fatal effect later in the test.
Those checks could either kept in the `acquireMaxNumberOfNodePorts` test or done in the lambda in one specific test case. The lambda would then take three parameters instead of one